src/CMakeLists.txt

```diff
@@ -115,6 +116,19 @@ endif()
 if (BUILD_MODULES)
     add_subdirectory(modules)
 endif()
+if (BUILD_FUZZER)

```

Comment:
Move this whole block before line 108 (before the `add_subdirectory` calls) otherwise the compile options are not applied the way you intended.